invenio_rdm_records/resources/iiif.py

```diff
@@ -0,0 +1,379 @@
+class IIIFResource(ErrorHandlersMixin, Resource):
+    """IIIF resource."""
+
+    def __init__(self, config, service):
+        """Instantiate resource."""
+        super().__init__(config)
+        self.service = service
+
+    @cached_property
+    def proxy(self):
+        """IIIF Image Server proxy instance."""
+        if self.config.proxy_cls is not None:
+            return self.config.proxy_cls()
+
+    @staticmethod
+    def proxy_pass(f):
+        """Decorate a function to proxy the request to an Image Server if enabled."""
+
+        @wraps(f)
+        def _wrapper(self, *args, **kwargs):
+            if self.proxy:
+                res = self.proxy()
+                if res:
+                    return res, 200
+            return f(self, *args, **kwargs)
+
+        return _wrapper
+
+    def create_url_rules(self):
+        """Create the URL rules for the IIIF resource."""
+        routes = self.config.routes
+        return [
+            route("GET", routes["manifest"], self.manifest),
+            route("GET", routes["sequence"], self.sequence),
+            route("GET", routes["canvas"], self.canvas),
+            route("GET", routes["image_base"], self.base),
+            route("GET", routes["image_info"], self.info),
+            route("GET", routes["image_api"], self.image_api),
+        ]
+
+    def _get_record_with_files(self):
+        uuid = resource_requestctx.view_args["uuid"]
+        return self.service.read_record(uuid=uuid, identity=g.identity)
+
+    #
+    # IIIF Manifest - not all clients support content-negotiation so we need a
+    # full endpoint.
+    #
+    # See https://iiif.io/api/presentation/2.1/#responses on
+    # "Access-Control-Allow-Origin: *"
+    #
+    @cross_origin(origin="*", methods=["GET"])
+    @with_iiif_content_negotiation(IIIFManifestV2JSONSerializer)
+    @iiif_request_view_args
+    @response_handler()
+    @proxy_pass.__func__
+    def manifest(self):
+        """Manifest."""
+        return self._get_record_with_files().to_dict(), 200
+
+    @cross_origin(origin="*", methods=["GET"])
+    @with_iiif_content_negotiation(IIIFSequenceV2JSONSerializer)
+    @iiif_request_view_args
+    @response_handler()
+    @proxy_pass.__func__
+    def sequence(self):
+        """Sequence."""
+        return self._get_record_with_files().to_dict(), 200
+
+    @cross_origin(origin="*", methods=["GET"])
+    @with_iiif_content_negotiation(IIIFCanvasV2JSONSerializer)
+    @iiif_request_view_args
+    @response_handler()
+    @proxy_pass.__func__
+    def canvas(self):
+        """Canvas."""
+        uuid = resource_requestctx.view_args["uuid"]
+        key = resource_requestctx.view_args["file_name"]
+        file_ = self.service.get_file(uuid=uuid, identity=g.identity, key=key)
+        return file_.to_dict(), 200
+
+    @cross_origin(origin="*", methods=["GET"])
+    @with_iiif_content_negotiation(IIIFInfoV2JSONSerializer)
+    @iiif_request_view_args
+    @response_handler()
+    @proxy_pass.__func__
```

Comment:
Out of curiosity, why is the `__func__` part needed?

Reply:
Because it's a `@staticmethod`, so it's "wrapped". Keeping it without the `@statimethod` looke more problematic/dangerous in case someone called this on an instance of the class.